### PR TITLE
Add component index to home pages

### PR DESCRIPTION
**Pull request by devongovett**
This adds an index of components to the home pages of React Aria and React Spectrum, with a grid of cards containing visual representations in a style similar to the anatomy diagrams.

To do

- [x] Make the categories more organized
- [x] Finish adding all the remaining components - decided not to add some minor ones
- [x] Figure out how to represent things that aren't components, e.g. interaction hooks – usePress, useMove, useLongPress, useKeyboard, useHover, useFocus, useFocusWithin, useFocusRing, FocusScope

### CODE REVIEW

**Comment by devongovett** [comment]
* [React Spectrum](https://reactspectrum.blob.core.windows.net/reactspectrum/bbca8188b93e200b44c0f0045bf6f399330ab5e4/docs/react-spectrum/index.html)
* [React Aria](https://reactspectrum.blob.core.windows.net/reactspectrum/bbca8188b93e200b44c0f0045bf6f399330ab5e4/docs/react-aria/index.html)

File: packages/dev/docs/src/docs.css [hunk 1]
```diff
@@ -889,3 +889,33 @@ h2.sectionHeader {
 .popover {
   @mixin small-propTable;
 }
+
+.index {
+  h3 {
+    margin-bottom: var(--spectrum-global-dimension-size-200) !important;
```
Comment (snowystinger) [inline packages/dev/docs/src/docs.css:895]: I can't review this without saying, "I see this" =P

**Review by snowystinger — APPROVED** [review]
They look nice :)

File: packages/dev/docs/src/docs.css (hunk 1 above)
Comment (devongovett) [reply packages/dev/docs/src/docs.css]: I was able to get rid of it :)

**Review by reidbarber — APPROVED** [review]
LGTM, this is awesome! Only comment would be that maybe we should think about removing the hero image so these get more attention.

File: packages/dev/docs/pages/react-spectrum/index.mdx [hunk 2]
```diff
@@ -68,3 +108,319 @@ A React implementation of Spectrum, Adobe’s design system.
...
+
+<ExampleCard
+  url="DialogTrigger.html"
+  title="DialogTrigger"
+  description="Displays a dialog in a popover, modal, or tray in response to a trigger.">
+  <Popover />
+</ExampleCard>
+
+<ExampleCard
+  url="TooltipTrigger.html"
```
Comment (LFDanLu) [inline packages/dev/docs/pages/react-spectrum/index.mdx:269]:
```suggestion
  url="Tooltip.html"
```

File: packages/dev/docs/pages/react-spectrum/index.mdx [hunk 3]
```diff
@@ -68,3 +108,319 @@ A React implementation of Spectrum, Adobe’s design system.
...
+</ExampleCard>
+
+</section>
+
+### Navigation
+
+<section className={styles.cardGroup}>
+
+<ExampleCard
+  url="TabList.html"
```
Comment (LFDanLu) [inline packages/dev/docs/pages/react-spectrum/index.mdx:372]:
```suggestion
  url="Tabs.html"
```

**Review by LFDanLu — COMMENTED** [review]
Couple of broken links, I'll commit

**Review by LFDanLu — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**